### DISCUSSION TITLE
Virus in releases

### DISCUSSION THREAD

**Discussion by ch4r1i3b** [T-Troll/alienfx-tools#49] [General]
Hi T-Troll
First of all, I would like to contact you privately but I couldn't find your email address.
Your alienfx-tools are awesome. I installed Linux in my Alienware and your tools are the only ones I can use to control my keyboard lights from a Windows 10 VM inside my Linux.
I've been updating your tools regularly and in the last releases I noticed they are infected.
I uploaded all your alienfx-tools.msi release versions to Virustotal with the following verdicts:

3.2.3:  Antivirus: MaxSecure -> Virus:Trojan.Malware.300983.susgen (maybe a false positive)
3.3.0: Clean
4.0.0: Clean
4.0.1: Clean
4.0.2: Clean
4.0.3:  Antivirus: Ikarus -> Virus:Trojan.Win64.Bazar (maybe a false positive)
4.1.0:  Antivirus: Ikarus -> Virus:Trojan.Win64.Bazar (maybe a false positive)
4.2.0:  Antivirus: Ikarus -> Virus:Trojan.Win64.Bazar (maybe a false positive)
**4.2.1:  30 security vendors flagged this file as malicious**
**4.2.2:  31 security vendors flagged this file as malicious**

As you can see, the last 2 releases are infected.
Please check them by yourself: (https://www.virustotal.com)

I hope this is of some help for you.

Regards!

**Comment by T-Troll** [comment]
Thank you for contact me. If you still need my e-mail, i'll send you one here or into some of your repo issues.

And yes, you're right. All releases after 4.2.1 WILL BE detected as a virus.
Do you know why (in fact, answer is in readme)?

I just DENY the idea to pay MS $400 per year for driver signing certificate. This tool is FREE and will be FREE.
But for my community, it is not a good idea to switch to Test Mode - it's not so easy and decrease overall system security.

So i use dirty hack to load my kernel driver without any signed certificate - and this detected as a rootkit (in fact, it is).

100% of my source code is available for review, you can build it yourself and have the same applications and distributing packages.
So it's not a virus. But will be detected as a virus, because MS want money.

Point.

PS: I looking for a guys who will provide free code sign certs for open source. If it will be - i sign the driver and remove the hack.
PPS: Hack source code is open source too, see credits of [alienfan](https://github.com/T-Troll/alienfan-tools)
PPPS: I think, i will write this case in big red letters at the top of readme at next release.

**Comment by T-Troll** [comment]
Funny... Right after i integrate kernel loader code into apps, VirusTotal, as well as my local AVs, don't detect any threat into package anymore.

That's made me worried about current AVs, in fact.

**Reply by ch4r1i3b** [reply]
You're right! I tested all from 4.2.3 on and all of them are clean!
Weird.....!

**Comment by ch4r1i3b** [comment]
Hi!
Ok, thank you for the explanation. 
I'm a security guy, so I always use to check anything I test or install. 
I think you've done a very good work so I wanted to warn you about it.
I hope you can solve this soon.
Regards!

**Comment by T-Troll** [comment]
Thank you for warning, it was predictable a bit, but nice to have a confirmation. Feel free to share if you found something later on.

I add a disclaimer at the beginning of readme, and also describe how to stop AV to complain (but you can only use Test Mode in this case, decreasing overall security).

AFAIK some guys (including Google and other) now design a code signing service for open source projects, but it doesn't operate for now. So let's see.

Anyway, as i said, 100% of my code is available in sources, so i don't afraid about any audit - no virus or something harmful inside.
